pkg/logql/metrics.go

```diff
@@ -133,3 +144,34 @@ func QueryType(query string) (string, error) {
 		return "", nil
 	}
 }
+
+// tagsToKeyValues converts QueryTags to form that is easy to log.
+// e.g: `Source=foo,Feature=beta` -> []interface{}{"source", "foo", "feature", "beta"}
+// so that we could log nicely!
+// If queryTags is not in canonical form then its completely ignored (e.g: `key1=value1,key2=value`)
+func tagsToKeyValues(queryTags string) []interface{} {
+	toks := strings.FieldsFunc(queryTags, func(r rune) bool {
+		return r == ','
+	})
+
+	vals := make([]string, 0)
+
+	for _, tok := range toks {
+		val := strings.FieldsFunc(tok, func(r rune) bool {
+			return r == '='
+		})
+
+		if len(val) != 2 {
```

Comment:
Why is this here?

Reply:
To check our header value is in canonical form `key1=key2, key3=key4`. Once you split by =, the number of tokens should be 2.

Reply:
How about naming it `pair` or `keyValPair`?